crates/rust-analyzer/src/config.rs

```diff
@@ -153,7 +153,8 @@ config_data! {
         /// Whether to show `can't find Cargo.toml` error message.
         notifications_cargoTomlNotFound: bool      = "true",
         /// Enable Proc macro support, `#rust-analyzer.cargo.loadOutDirsFromCheck#` must be
-        /// enabled.
+        /// enabled. This has nothing to do with "proc macro not expanded" warnings. See the
```

Comment:
> This has nothing to do with "proc macro not expanded" warnings

That doesn't seem right to me? The "proc macro not expanded" warnings are emitted when we encounter proc macro calls, but this setting is disabled, so they have a lot to do with each other.

Reply:
The warnings seem to be shown whether this setting is enabled or not. That is what I was attempting to capture here. I hadn't actually heard of a "proc macro" before this (though I now have some idea what they are :slightly_smiling_face: ). I expect a lot of users who see this message when using rust-analyzer won't really know either. I'm still not sure what this setting is for, but I guess it is only required if you are writing macros and have the [corresponding flag](https://doc.rust-lang.org/reference/procedural-macros.html) set in your crate. Is that correct?

Reply:
>  I hadn't actually heard of a "proc macro" before this

If you've used `serde`, that was a proc macro. They're called procedural macros because they're Rust functions that are called with bits of text and return other bits of Rust code. That's unlike declarative macros (or "macros by example"), which define input templates and rules for how they need to be transformed.

A proc macro lives in a Rust crate that needs to be compiled, then loaded at run-time by the compiler (so that it's able to call the function in there). We're not the compiler, but we do the same thing to expand procedural macros, but only if `rust-analyzer.procMacro.enable` (because of https://github.com/rust-analyzer/rust-analyzer/issues/6448).

Reply:
Thanks, you are right. The warnings *do* go away with this enabled. I'm not sure why this didn't work for me the first time I tried it. I probably had a typo in the nvim lua settings. So this change is definitely wrong. Shall I change it to to something along the lines of "If you are seeing 'proc macro not expanded' warnings, enabling this property should stop them. Alternatively you can disable the `unresolved-proc-macro` diagnostic"?

Reply:
> Shall I change it to to something along the lines of "If you are seeing 'proc macro not expanded' warnings, enabling this property should stop them. Alternatively you can disable the unresolved-proc-macro diagnostic"?

That sounds correct. It might be too verbose, but I'm not sure how verbose the descriptions usually are.